src/dmd/glue.d

```diff
@@ -1163,8 +1163,67 @@ void FuncDeclaration_toObjFile(FuncDeclaration fd, bool multiobj)
 
     if (fd.isCrtCtorDtor & 1)
         objmod.setModuleCtorDtor(s, true);
+
     if (fd.isCrtCtorDtor & 2)
-        objmod.setModuleCtorDtor(s, false);
+    {
+        bool mitigation = true;
+        /*
+            https://issues.dlang.org/show_bug.cgi?id=22520
+
+            Apple radar: https://openradar.appspot.com/FB9733712
+
+            Apple deprecated the mechanism used to implement `crt_constructor`
+            on MacOS Monterey. This works around that by generating a new function
+            (crt_destructor_thunk_NNN, run as a constructor) which registers
+            the destructor-to-be using __cxa_atexit()
+
+            This workaround may need a further look at when it comes to
+            shared library support, however there is no bridge for
+            that spilt milk to flow under yet.
+
+            This relies on the Itanium ABI so is portable to any
+            platform it, if needed.
+        */
+        if(mitigation)
```

Comment:
I was talking about this branching, perhaps use `static if` to avoid uncovered lines of unreachable code.

Reply:
Code is visibly not unreachable though, and if it gets flagged then that's just a limitation in the coverage report.

(That this bool is always true just tells me that this PR is still WIP and should really be a draft)